src/Tooltip/Tooltip.tsx

```diff
@@ -1,231 +1,143 @@
-  }, [tipContainer, tooltipPosition])
+  const arrowRef = useRef(null)
 
-  useEffect(() => {
-    handleTipViewport()
-    setChildEl(document.getElementById(randomId))
-  }, [])
+  const variantValue = tooltipVariants[variant]
+  const bgColor = bgColorProp ? theme.colors[bgColorProp] : theme.colors.custard
 
-  useEffect(() => {
-    setTooltipPosition(position)
-  }, [position])
-
-  const arrowSize = 18
-
-  const calculateTooltipPosition = (): { top: number; left: number } => {
-    if (!childEl) return { top: 0, left: 0 }
-    if (!tipContainer.current) return { top: 0, left: 0 }
-
-    const childPosition = childEl.getBoundingClientRect()
-    const tipDimensions = tipContainer.current.getBoundingClientRect()
-
-    let top = 0
-    let left = 0
-
-    switch (tooltipPosition) {
-      case 'top':
-        top =
-          childPosition.top + window.scrollY - tipDimensions.height - arrowSize
-        left =
-          childPosition.left +
-          window.scrollX +
-          childPosition.width / 2 -
-          tipDimensions.width / 2
-        break
-      case 'bottom':
-        top = childPosition.bottom + window.scrollY + arrowSize
-        left =
-          childPosition.left +
-          window.scrollX +
-          childPosition.width / 2 -
-          tipDimensions.width / 2
-        break
-      case 'left':
-        left =
-          childPosition.left + window.scrollX - tipDimensions.width - arrowSize
-        top =
-          childPosition.top +
-          window.scrollY +
-          childPosition.height / 2 -
-          tipDimensions.height / 2
-        break
-      case 'right':
-        left = childPosition.right + window.scrollX + arrowSize
-        top =
-          childPosition.top +
-          window.scrollY +
-          childPosition.height / 2 -
-          tipDimensions.height / 2
-        break
-      default:
-        break
-    }
-    return { top, left }
-  }
-
-  useEffect(() => {
-    if (childEl) {
-      setTooltipCoords(calculateTooltipPosition())
-    }
-  }, [
-    position,
-    window.scrollY,
-    tipContainer.current,
-    childEl,
-    childEl?.getBoundingClientRect().top,
-    childEl?.getBoundingClientRect().left,
-  ])
+  const randomId = crypto.randomUUID()
 
-  useEventListener({
-    eventName: 'resize',
-    callback: debounce(handleTipViewport, 100),
-    ref: documentRef,
+  const { refs, floatingStyles, context } = useFloating({
+    open: showTip,
+    placement: position,
+    onOpenChange: setShowTip,
+    middleware: [
+      offset(ARROW_HEIGHT + GAP),
+      flip(),
+      shift(),
+      arrow({ element: arrowRef, padding: 14 }),
+    ],
+    whileElementsMounted: autoUpdate,
   })
 
-  useEventListener({
-    eventName: 'scroll',
-    callback: debounce(handleTipViewport, 100),
-    ref: documentRef,
-  })
+  const hover = useHover(context, { move: false })
+  const focus = useFocus(context)
+  const dismiss = useDismiss(context)
+  const role = useRole(context, { role: 'tooltip' })
```

Comment:
[hover](https://floating-ui.com/docs/useHover): Opens the floating element while hovering over the reference element, like CSS :hover.
[focus](https://floating-ui.com/docs/useFocus): Opens the floating element while the reference element has focus, like CSS :focus.
[dismiss](https://floating-ui.com/docs/useDismiss): Closes the floating element when a dismissal is requested — by default, when the user presses the escape key or outside of the floating element with their pointer.
[role](https://floating-ui.com/docs/useRole): Adds base screen reader props to the reference and floating elements for a given `role`.
  - We could allow this to be configurable but I don't think it'll be needed